cms/djangoapps/contentstore/views/tests/test_item.py

```diff
@@ -1421,6 +1439,69 @@ def test_create_non_deprecated_problems(self):
         template_display_names = [template['display_name'] for template in templates[0]['templates']]
         self.assertEqual(template_display_names, ['Annotation', 'Poll', 'Survey'])
 
+    def test_deprecated_no_advance_component_button_with_admin_settings(self):
+        """
+        Test that there will be no `Advanced` button on unit page if units are
+        deprecated from the admin settings `XBlockDeprecatedAdvancedComponentConfig`
+        provided that they are the only modules in `Advanced Module List`
+        """
+        model = XBlockDeprecatedAdvancedComponentConfig.objects.create(disabled_blocks='poll survey', enabled=True)
+        model.save()
+        self._test_no_advance_component_button()
+
+    def test_cannot_create_deprecated_problems_with_admin_settings(self):
+        """
+        Test that we can't create problems if they are deprecated from the admin settings
+        """
+        model = XBlockDeprecatedAdvancedComponentConfig.objects.create(disabled_blocks='poll survey', enabled=True)
+        model.save()
+
+        self._test_cannot_create_deprecated_problems()
+
+    def test_create_non_deprecated_problems_with_admin_settings(self):
+        """
+        Test that we can create problems if they are in `Advanced Module List`
+        and not deprecated from admin settings
+        """
+        model = XBlockDeprecatedAdvancedComponentConfig.objects.create(disabled_blocks='poll', enabled=True)
+        model.save()
+
+        self.course.advanced_modules.extend(['annotatable', 'poll', 'survey'])
+        templates = get_component_templates(self.course)
+        button_names = [template['display_name'] for template in templates]
+        self.assertIn('Advanced', button_names)
+        self.assertEqual(len(templates[0]['templates']), 2)
+        template_display_names = [template['display_name'] for template in templates[0]['templates']]
+        self.assertEqual(template_display_names, ['Annotation', 'Survey'])
+
+    @patch('django.conf.settings.DEPRECATED_ADVANCED_COMPONENT_TYPES', ["poll"])
+    def test_deprecated_from_both_settings_file_and_admin_settings(self):
```

Comment:
We shouldn't worry about this here since this logic has now been moved to the `xblock_django` app. You should test that the class method returns the right list of block types whether the types are in settings file or in the config model.